applications/serial_lte_modem/src/slm_at_commands.c

```diff
@@ -270,10 +271,13 @@ static int handle_at_datactrl(enum at_cmd_type cmd_type)
 		if (ret) {
 			return ret;
 		}
-		if (datamode_time_limit < DATAMODE_TIME_LIMIT_MAX &&
-		    datamode_size_limit < DATAMODE_SIZE_LIMIT_MAX) {
-			datamode_size_limit = size_limit;
-			datamode_time_limit = time_limit;
+		if (size_limit < DATAMODE_SIZE_LIMIT_MAX && time_limit < DATAMODE_TIME_LIMIT_MAX) {
+			if (verify_datamode_control(size_limit, time_limit)) {
```

Comment:
The check of DATAMODE_SIZE_LIMIT_MAX/DATAMODE_TIME_LIMIT_MAX
can also be put inside verify_datamode_control

Reply:
Taken with gratitude